lib/vue/src/components/input/CInputExpressions.vue

```diff
@@ -0,0 +1,202 @@
+<template>
+  <div
+    class="position-relative"
+  >
+    <c-ace-editor
+      v-model="editorValue"
+      :initializeEditor="editorInit"
+      v-bind="{ ...$attrs, ...$props }"
+      v-on="$listeners"
+    />
+  </div>
+</template>
+
+<script>
+import CAceEditor from './CAceEditor.vue'
+
+export default {
+  components: {
+    CAceEditor
+  },
+
+  props: {
+    value: {
+      type: String,
+      default: '',
+    },
+
+    lang: {
+      type: String,
+      default: 'text',
+    },
+
+    height: {
+      type: String,
+      default: '80px',
+    },
+
+    showLineNumbers: {
+      type: Boolean,
+      default: false,
+    },
+
+    fontSize: {
+      type: String,
+      default: '14px',
+    },
+
+    border: {
+      type: Boolean,
+      default: true,
+    },
+
+    showPopout: {
+      type: Boolean,
+      default: false,
+    },
+
+    readOnly: {
+      type: Boolean,
+      default: false,
+    },
+
+    highlightActiveLine: {
+      type: Boolean,
+      default: false,
+    },
+
+    showPrintMargin: {
+      type: Boolean,
+      default: false,
+    },
+
+    suggestionTree: {
+      type: Object,
+      default: () => ({})
+    }
+  },
+
+  computed: {
+    editorValue: {
+      get () {
+        return this.value
+      },
+
+      set (value = '') {
+        this.$emit('update:value', value)
+      },
+    },
+  },
+
+  methods: {
+    editorInit (editor) {
+      require('brace/mode/javascript')
+
+      require('brace/theme/chrome')
+      require('brace/ext/language_tools')
+
+      editor.setOptions({
```

Comment:
If you provide props to the c-ace-editor this init function can be avoided.
That would keep the init logic only on the c-ace-editor which would be nice.
This wrapper would just define the appropriate suggestions (parsed from the suggestion tree prop)